qa/L0_lifecycle/lifecycle_test.py

```diff
@@ -2917,6 +2917,62 @@ def test_concurrent_model_load_unload(self):
             for model_name in model_names:
                 self.assertEqual(is_load, triton_client.is_model_ready(model_name))
 
+    def test_concurrent_same_model_load_unload_stress(self):
+        model_name = "identity_zero_1_int32"
+        num_threads = 16
+        num_iterations = 1024
+        try:
+            triton_client = grpcclient.InferenceServerClient(
+                "localhost:8001", verbose=True
+            )
+        except Exception as ex:
+            self.assertTrue(False, "unexpected error {}".format(ex))
+        load_unload_exceptions = {"load_before_unload_finish_count": 0}
+
+        def _load_unload():
+            for i in range(num_iterations):
+                try:
+                    triton_client.load_model(model_name)
+                except InferenceServerException as ex:
+                    # Acceptable for an unload to happen after a load completes, but
+                    # before the load can verify its load state.
+                    fail_reasons = [
+                        "unexpected miss in global map",
```

Comment:
I don't think I've ever seen this error, is this a valid user-facing error?

Reply:
Those errors are from https://github.com/triton-inference-server/core/blob/main/src/model_repository_manager.cc#L663-L706.

For model load, notice the `mu_` that ensure atomic operation is release briefly between the actual load completion and validation. If the validation fails, the errors could be printed. i.e. an unload of the same model happens while the `mu_` is briefly released. Thus, the check is there to prevent intermittent failure.
(I was never able to produce any load error, maybe the gap is simply too small, but the probability is not zero I think.)